File: rainfalldeciles.py
import pandas as pd
import matplotlib.pyplot as plt

# Read the rainfall data from CSV file
data = pd.read_csv('rainfall_data.csv')

# Calculate the deciles
deciles = data['Rainfall'].quantile([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9])

# Display deciles in text
print("Deciles of Rainfall:")
print(deciles)

# Create a bar plot of deciles
plt.bar(deciles.index, deciles.values)
plt.xlabel('Decile')
plt.ylabel('Rainfall')
plt.title('Deciles of Rainfall')
plt.show()

File: rainfalldeciles.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

class Program
{
    static void Main()
    {
        // Read CSV file
        string[] lines = File.ReadAllLines("rainfall_data_fortran_sorted.csv");

        // Parse the rainfall data and sort
        List<double> rainfallData = lines.Select(line => double.Parse(line)).ToList();
        rainfallData.Sort();

        // Calculate and print deciles
        for (int i = 1; i <= 10; i++)
        {
            double decile = CalculateDecile(rainfallData, i);
            Console.WriteLine($"Decile {i}: {decile}");
        }
    }

    static double CalculateDecile(List<double> data, int decile)
    {
        int index = (int)Math.Ceiling((data.Count * decile) / 10.0) - 1;
        return data[index];
    }
}

